tensorbay/cli/log.py

```diff
@@ -24,32 +28,138 @@
 """
 
 
+class LogCommits:  # pylint: disable=too-few-public-methods
+    """This class defines the structure of logging commits.
+
+    Arguments:
+        all_commits: The all commits PagingList from valid branches.
+
+    """
+
+    def __init__(self, all_commits: List[PagingList[Commit]]):
+        # Sorted commits paging list by date.
+        self._sorted_commits: List[PagingList[Commit]] = sorted(
+            all_commits, key=lambda x: x[0].committer.date
+        )
+        self._keys = [commits[0].committer.date for commits in self._sorted_commits]
+
+    def _merge(self, latest_commit: Commit) -> bool:
+        if len(self._sorted_commits) <= 1:
+            return False
+
+        date = latest_commit.committer.date
+        commit_id = latest_commit.commit_id
+        # Determine whether to merge branches when different commits have the same timestamp.
+        for commits in islice(reversed(self._sorted_commits), 1, None):
+            commit = commits[0]
+            # Different timestamp, don't merge.
```

Comment:
Traverse all the commits with the same timestamp, if the commit id is the same as the lastest commit, then merge the branch where the latest commit is located.